org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/RenameHandlerTest.java

```diff
@@ -234,18 +235,18 @@ public void testRenameTypeWhithResourceChanges() throws JavaModelException, BadL
 
 		WorkspaceEdit edit = getRenameEdit(cu, pos, "Newname");
 		assertNotNull(edit);
-		List<Either<ResourceChange, TextDocumentEdit>> resourceChanges = edit.getResourceChanges();
+		List<Either<TextDocumentEdit, ResourceOperation>> resourceChanges = edit.getDocumentChanges();
 
 		assertEquals(resourceChanges.size(), 3);
 
-		Either<ResourceChange, TextDocumentEdit> change = resourceChanges.get(2);
-		ResourceChange resourceChange = change.getLeft();
-		assertEquals(JDTUtils.toURI(cu), resourceChange.getCurrent());
+		Either<TextDocumentEdit, ResourceOperation> change = resourceChanges.get(2);
+		RenameFile resourceChange = (RenameFile) change.getRight();
+		assertEquals(JDTUtils.toURI(cu), resourceChange.getOldUri());
 		assertEquals(JDTUtils.toURI(cu).replace("E", "Newname"), resourceChange.getNewUri());

```

Comment:
assertEquals(JDTUtils.toURI(cu).replace("E.java", "Newname.java"), resourceChange.getNewUri());

If the workspace name contains "E", the test will fail.

Reply:
@snjeza  Add test cases for failure situation.

Reply:
I agree with @snjeza, the regexp in replace() call needs to be more specific